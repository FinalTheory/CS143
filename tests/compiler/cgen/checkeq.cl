class Main inherits IO
{
  obj:Object;
  main(): Object {
        { if self = obj
            then out_string ("self = obj\n")
            else out_string ("self != obj\n")
          fi;
          if 1 = 0
            then out_string ("1 = 0\n")
            else out_string ("1 != 0\n")
          fi;
          if true = false
            then out_string ("true = false\n")
            else out_string ("true != false\n")
          fi;
          if "abc" = "bbc"
            then out_string ("\"abc\" = \"bbc\"\n")
            else out_string ("\"abc\" != \"bbc\"\n")
          fi;
      }
  };
};
