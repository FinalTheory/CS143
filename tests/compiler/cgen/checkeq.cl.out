SPIM Version 6.5 of January 4, 2003
Copyright 1990-2003 by James R. Larus (larus@cs.wisc.edu).
All Rights Reserved.
See the file README for a full copyright notice.
Loaded: /usr/class/cs143/cool/lib/trap.handler
self != obj
1 != 0
true != false
"abc" != "bbc"
COOL program successfully executed
Stats -- #instructions : 324
         #reads : 67  #writes 29  #branches 76  #other 152
